linkerd/metrics/src/counter.rs

```diff
@@ -15,75 +15,70 @@ use super::prom::{FmtLabels, FmtMetric};
 /// [`rate()`]: https://prometheus.io/docs/prometheus/latest/querying/functions/#rate()
 /// [`irate()`]: https://prometheus.io/docs/prometheus/latest/querying/functions/#irate()
 /// [`resets()`]: https://prometheus.io/docs/prometheus/latest/querying/functions/#resets
-#[derive(Copy, Clone, Debug, Default, Eq, PartialEq)]
-pub struct Counter(u64);
-
-/// Largest `u64` that can fit without loss of precision in `f64` (2^53).
```

Comment:
I think we need to retain this behavior for prometheus compatibility. I think that rather than `fetch_add`, we need to change this code to do a load, 52-bit wrapping add, and try to CAS the new value back. This probably needs to be a CAS loop.